src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -259,83 +259,121 @@ private JsonPropertyInfo GetPropertyWithUniqueAttribute(Type attributeType, Dict
 
         public JsonPropertyInfo GetProperty(ReadOnlySpan<byte> propertyName, ref ReadStackFrame frame)
         {
-            JsonPropertyInfo info = null;
-
             // Keep a local copy of the cache in case it changes by another thread.
             PropertyRef[] localPropertyRefsSorted = _propertyRefsSorted;
 
+            ulong key = GetKey(propertyName);
+
             // If there is an existing cache, then use it.
             if (localPropertyRefsSorted != null)
             {
-                ulong key = GetKey(propertyName);
-
                 // Start with the current property index, and then go forwards\backwards.
                 int propertyIndex = frame.PropertyIndex;
 
                 int count = localPropertyRefsSorted.Length;
                 int iForward = Math.Min(propertyIndex, count);
                 int iBackward = iForward - 1;
 
-                while (iForward < count || iBackward >= 0)
+                for (;;)
```

Comment:
Do we have some evidence to suggest that this forward/backwards search will be beneficial in practice, for perf?